lightning/src/ln/channelmanager.rs

```diff
@@ -1380,6 +1393,23 @@ macro_rules! handle_error {
 	}
 }
 
+macro_rules! update_maps_on_chan_removal {
+	($self: expr, $short_to_id: expr, $channel: expr) => {
+		if let Some(short_id) = $channel.get_short_channel_id() {
+			$short_to_id.remove(&short_id);
+		} else {
+			// If the channel was never confirmed on-chain prior to its closure, remove the
+			// outbound SCID alias we used for it from the collision-prevention set. While we
+			// generally want to avoid ever re-using an outbound SCID alias across all channels, we
+			// also don't want a counterparty to be able to trivially use up all our memory by
+			// simply opening a billion channels with us which are closed before we ever reach the
+			// funding stage.
```

Comment:
Does this actually prevent that, though, if this map is inserted into on `open_channel`?

Reply:
Right, I guess I should rephrase, I mean this prevents us *leaking*, but of course as long as the channels are pending we'll store channel info for them.